Dangerfile

```diff
@@ -22,13 +22,25 @@ changed_files = git.diff.stats[:files]
 excluded_changed_files = changed_files.select { |key| EXCLUSIONS.any? { |exclusion| key.include?(exclusion) } }
 filtered_changed_files = changed_files.reject { |key| EXCLUSIONS.any? { |exclusion| key.include?(exclusion) } }
 
+# fetch master for diff comparison
+`git fetch origin master`
+
+# get branch name
+branches = `git branch`
+current_branch = branches.match(/\*\s+([^\s]+)/m)[1]
+
 # ignores whitespace for the purpose of determining lines of code changed
-changes = `git diff master... -w --stat`.split("\n")
-lines_of_code = changes.sum(0) do |change|
-  if change == changes.last || EXCLUSIONS.any? { |exclusion| change.match(exclusion) }
-    0
-  else
-    change.match(/\|\s+(\d+)/)[1].to_i
+diff_changes = `git diff master..#{current_branch} -w --stat`.split("\n")
+
+if diff_changes.empty?
```

Comment:
what's this case where the diff is empty?

Reply:
I was coming up with some situations locally where it would sometimes be blank. I'm really uncertain whether or not that will happen in production or if it is just one of the issues with testing locally. In any case, if it comes back empty I want it to fail back to the way it was originally calculating the LOC changed. Better to have it count whitespace in LOC changed than to have it pass on something that has too many lines.

Reply:
I think it's worth trying to discover what causes that case.  If this is inconsistent I don't know how helpful it will be if it sometimes flags PRs and sometimes doesn't because it did/didn't count whitespace.
I think catching fail cases is a good idea though, it'll help the dangerbot to run successfully instead of being blank.

Reply:
It could be that you need to use a [three dot diff](https://stackoverflow.com/questions/462974/what-are-the-differences-between-double-dot-and-triple-dot-in-git-com) instead of two.